Gems/AtomTressFX/CMakeLists.txt

```diff
@@ -5,3 +5,159 @@
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 #
 #
+
+set(GemUUID d4a3e3d6b9b14e12982c774f35f48a7c)
+
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
+
+    # Declare ATOMTRESSFX_EDITOR only when tools pipeline is built
+    # This is in order to allow AzToolsFramework to properly link across
+    # platforms that support editor mode, while in the rest, not include
+    # this module and avoide compiling the editor components.
+    set_source_files_properties(
+        Code/HairModule.cpp
+        Code/Components/EditorHairComponent.h
+        Code/Components/EditorHairComponent.cpp
+        PROPERTIES
+            COMPILE_DEFINITIONS
+                ATOMTRESSFX_EDITOR
+    )
+
+    ly_add_target(
+        NAME AtomTressFX.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            Hair_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Code
+                External
+                External/Code/src
+            PUBLIC
+                Code/Include
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzCore
+                AZ::AzToolsFramework
+                Gem::LmbrCentral
+                Gem::Atom_RHI.Public
+                Gem::Atom_RPI.Public
+                Gem::Atom_Feature_Common.Static
+                Gem::AtomLyIntegration_CommonFeatures.Static
+                Gem::EMotionFXStaticLib
+    )
+
+    ly_add_target(
+        NAME AtomTressFX ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+        NAMESPACE Gem
+        FILES_CMAKE
+            Hair_shared_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Code/AssetPipeline
+                Code
+                External
+                External/Code/src
+            PUBLIC
+                Code/Include
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzCore
+                AZ::AzToolsFramework
+                Gem::Atom_Utils.Static
+                Gem::EMotionFXStaticLib
+                Gem::AtomTressFX.Static
+    )
+
+else()
+
+    ly_add_target(
```

Comment:
So this is not the usual strategy, but its not technically wrong per-se.

f(PAL_TRAIT_BUILD_HOST_TOOLS) means "if we are building for a system that can have tools" - so for example, mac, linux, windows contrast with for example, ios, android, etc, which you do not deploy tools like Editor to.

So this strategy in a nutshell seems to be
```
if (we are on a system that can have tools)
   Declare a static library called AtomTressFX.Static
   Declare a Gem moule called AtomTressFx
else (if we are on a system that has no tools, like, for example, ios)
   Declare a static library called AtomTressFX.Static with the same files but it also depends on a few extra libraries
   Declare a gem module called AtomTressFx again, with the same files, and same dependencies
endif


Associate the "AtomTressFX module" with both clients and servers

if (we are on a system that can have tools, again!)
   create a static library for builders
   Create a Gem MOdule for Builders (becuase it ends with .Builders its automatically associated)
   
(make no editor plugin)
endif()

```


this does not seem right?  Contrast with the normal way of doing it:
```
create a static library for all the shared functionality (atomTressFX.Static)
create a gem module that is the client side code for all game clients (AtomTressFX) 
    make it depend on the static library

Associate the client side gem with Clients (ly_create_alias...)

if (we are on a host system that can have tools)
   also create a gem module that is for tools (AtomTressFX.editor)
      depend on the common static library for common code
   optionally create a gem module that will only be loaded in asset processor (AtomTressFX.pipeline)
   
  associate the "editor" module with all tools (ly_create_alias(...Tools ... .Editor)
  opitonally associate the "builders" module with builders (ly_create_alias(...Builders ... Pipeline)
  (if you have no builder module you can associate the editor module with builders)
endif

Reply:
Seems like the only extra dependency added for the AtomTressFX.Static Target is AZ::AzToolsFramework. It seems like there isn't an AtomTressFX.Editor target added, but instead it glued on top of the static and runtime gem?

Reply:
Yes , its difficult for me to understand the intention here.   There might be a good reason for it but it kind of seems like something was missed in the tiered layout of the architecture here.  Or just a simple error.

Reply:
True, there is no intention (like in all other places you pointed out and I traced) to have a separated editor / tools module.
However, the existing module already contains the editor add-on, but the problem is that on platform without the capability to support editor this would be a problem.
For this reason I needed to introduce compilation flags and add / remove content based on the platform or else I would get warnings that are now treated as errors and will break builds of such platforms.
I consulted several people and this is the only viable solution that seemed to work with the given intent and configuration and as you said - it's not wrong in design, simply not the best design I would have liked.
@lawsonamzn , please let me know if this is a blocker for you since this unblocks compilation on all platforms without the need to create any dummy module or even worse - separate into a not-required editor module.

Reply:
As I Said, its not a blocker, its just not standard.
I assume by "Platforms without the capability to support editor" you mean, say, ios, android, etc, right?
Linux and Mac are now editor supported, and we don't really want to launch features without standard support for those.